src/proxy-builder.ts

```diff
@@ -10,17 +12,32 @@ export interface RequestProxy {
       password: string;
     };
   };
+  httpsAgent?: Agent;
 }
 
 export const configureProxy = (requestURL: string): RequestProxy => {
-  const url: URL = new URL(requestURL);
-  const env: string = getProxyEnv(url);
+  const requestURLObject: URL = new URL(requestURL);
+  const proxyUrl: string = getProxyEnv(requestURLObject);
 
   // short circuit if null
-  if (!env) return null;
+  if (!proxyUrl) return null;
 
   // parse proxy url
-  const { hostname, port, protocol, username, password } = new URL(env);
+  const { hostname, port, protocol, username, password } = new URL(proxyUrl);
+
+  // axios proxy implementation for https over http doesn't work. hence, this implementation
+  if (requestURLObject.protocol === "https:" && protocol === "http:") {
+    const agent = httpsOverHttp({
+      proxy: {
+        host: hostname,
+        port: parseInt(port),
+      },
+    });
+
+    return {
+      httpsAgent: agent,
+    };
+  }
```

Comment:
seems like issue after fixing the merge conflict

Reply:
No, issue isn't here, it's in our null checks in `./utils`. Its working as designed, just not for tunneling.